library/src/test/java/io/getstream/chat/android/client/api/ClientConnectionTests.kt

```diff
@@ -91,6 +93,22 @@ internal class ClientConnectionTests {
         verify(socket, times(1)).connect(user)
     }
 
+    @Test
+    fun `Should not connect when user is already set`() {
+
+        `when`(socket.events()).thenReturn(JustObservable(connectedEvent))
+
+        client = ChatClientImpl(
+            config,
+            api,
+            socket,
+            notificationsManager
+        )
+        client.setUser(user, token)
```

Comment:
Could we call with a listener and check that the proper method is called?

Reply:
Yes we can